app/services/solidus_bolt/payments/credit_sync_service.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module Payments
+    class CreditSyncService < BaseService
+      attr_reader :payment, :amount, :transaction_id, :refund_reason, :order
+
+      def initialize(payment:, amount:, transaction_id:)
+        @payment = payment
+        @amount = amount
+        @transaction_id = transaction_id
+        @refund_reason = Spree::RefundReason.return_processing_reason&.id
+        @order = payment.order
+        super
+      end
+
+      def call
+        return if Spree::Refund.find_by(transaction_id: transaction_id)
+
+        create_refund
+        order.recalculate
+      end
+
+      private
+
+      def create_refund
+        payload = {
+          payment_id: payment.id,
+          amount: amount.to_f / 100,
+          transaction_id: transaction_id,
+          refund_reason_id: refund_reason
+        }
+
+        if Gem.loaded_specs['solidus'].version >= Gem::Version.create('3.0')
```

Comment:
🤔 
Why you didn't use `Spree.solidus_gem_version`?

Reply:
fixed in 2e85755